src/nvim/eval.c

```diff
@@ -15850,11 +15850,10 @@ static void f_sign_getplaced(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       }
       if ((di = tv_dict_find(dict, "lnum", -1)) != NULL) {
         // get signs placed at this line
-        lnum = (linenr_T)tv_get_number_chk(&di->di_tv, &notanum);
-        if (notanum) {
+        lnum = tv_get_lnum(&di->di_tv);
+        if (lnum <= 0) {
```

Comment:
would be better to check -1 explicitly since that is the contract of `tv_get_number_chk`.

Reply:
Docstring says tv_get_lnum returns line number or 0 or 1.
https://github.com/neovim/neovim/blob/7d604a1e6158cda79f7ac485e0ef0e349014f8bf/src/nvim/eval/typval.c#L2761
Are you sure about that only -1 check is fine?